### ISSUE TITLE
Add support for modern browsers

### ISSUE THREAD

**Issue by compulim** [l12s/node-slack-client#23]
Modern browsers have Web Socket support and should be analogous to `ws` node package. For REST-ful calls, should consider moving the transport to `fetch`, which is supported in modern browsers, as well as some Node.js bridge packages.

To build a single JS file, could use Webpack for [umd](https://github.com/umdjs/umd) and BabelJS for [SystemJS](https://github.com/systemjs/systemjs). Both are very popular module loading mechanism. You can refer to `redux-devtools` for setting up umd build.

**Comment by l12s** [comment]
Yep, in theory (not at all tested yet), you can just pass in a separate transport and socketFn and compile everything up with browserify / webpack and it should just work.
